src/BuildScriptGenerator/Php/PhpPlatform.cs

```diff
@@ -173,20 +181,80 @@ public IEnumerable<string> GetDirectoriesToExcludeFromCopyToIntermediateDir(Buil
             return Array.Empty<string>();
         }
 
-        public string GetInstallerScriptSnippet(
-            BuildScriptGeneratorContext scriptGeneratorContext,
-            PlatformDetectorResult detectorResult)
-        {
-            return null;
-        }
-
         public string ResolveVersion(string versionToResolve)
         {
             var resolvedVersion = GetVersionUsingHierarchicalRules(versionToResolve);
             resolvedVersion = GetMaxSatisfyingVersionAndVerify(resolvedVersion);
             return resolvedVersion;
         }
 
+        public string GetInstallerScriptSnippet(
+            BuildScriptGeneratorContext context,
+            PlatformDetectorResult detectorResult)
+        {
+            if (_commonOptions.EnableDynamicInstall)
+            {
+                var installationScriptSnippetBuilder = new StringBuilder();
+
+                _logger.LogDebug("Dynamic install is enabled.");
+
+                // Install PHP
+                var phpVersion = detectorResult.PlatformVersion;
+                if (_phpInstaller.IsVersionAlreadyInstalled(phpVersion))
+                {
+                    _logger.LogDebug(
+                       "PHP version {version} is already installed. So skipping installing it again.",
+                       phpVersion);
+                }
+                else
+                {
```

Comment:
how about splitting the install PHP and install composer blocks as separate methods for easier reading?